### PR TITLE
feat(Routes): voice state endpoint

### PR DESCRIPTION
**Pull request by imnaiyar**
**Please describe the changes this PR makes and why it should be merged:**
 
**If applicable, please reference Discord API Docs PRs or commits that influenced this PR:**
 - https://github.com/discord/discord-api-docs/pull/7061

### CODE REVIEW

**Comment by imnaiyar** [comment]
Also I'm not sure if we should rename `GatewayVoiceState` type (possibly to `APIVoiceState` ?) now that it is possible to get it over http

**Comment by imnaiyar** [comment]
> > Also I'm not sure if we should rename `GatewayVoiceState` type (possibly to `APIVoiceState` ?) now that it is possible to get it over http
> 
> Yes, and deprecating the gateway alias

Did that if you wanna have a look

File: rest/v10/index.ts [hunk 1]
```diff
@@ -609,6 +609,15 @@ export const Routes = {
 		return `/voice/regions` as const;
 	},
 
+	/**
+	 * Route for:
+	 * - GET `/guilds/{guild.id}/voice-states/@me`
+	 * - GET `/guilds/{guild.id}/voice-states/{user.id}`
+	 */
+	voiceState(guildId: Snowflake, userId: Snowflake | '@me' = '@me') {
```
Comment (TAEMBO) [inline rest/v10/index.ts:617]: Shouldn't we be reusing the existing [`guildVoiceState()`](https://github.com/discordjs/discord-api-types/blob/main/rest/v10/index.ts#L793) function for this instead? That function uses the same route as this, and would only require additional JSDoc comments to indicate it now supports the GET method as well.

Comment (imnaiyar) [reply rest/v10/index.ts:617]: You're right, I missed this. Though voice state endpoints have been moved to a separate category altogether (moved from Guild) in the API documentation, so I'm wondering if we should still call it `guildVoiceState`

Comment (imnaiyar) [reply rest/v10/index.ts:617]: Ig it doesn't make much sense renaming it.

**Review by TAEMBO — APPROVED** [review]

**Review by almeidx — CHANGES_REQUESTED** [review]
Missing the REST types, which should be added to rest/voice.ts to match the Discord docs

The patch endpoint types are in rest/guild.ts, which should be moved to rest/voice.ts as well. Not sure if we'd have to add a re-export on the patch types in rest/guild.ts since users *can* import individual files too

**Comment by imnaiyar** [comment]
> Missing the REST types, which should be added to rest/voice.ts to match the Discord docs
> 
> The patch endpoint types are in rest/guild.ts, which should be moved to rest/voice.ts as well. Not sure if we'd have to add a re-export on the patch types in rest/guild.ts since users _can_ import individual files too

I'm not sure if we should re-export from

> Missing the REST types, which should be added to rest/voice.ts to match the Discord docs
> 
> The patch endpoint types are in rest/guild.ts, which should be moved to rest/voice.ts as well. Not sure if we'd have to add a re-export on the patch types in rest/guild.ts since users _can_ import individual files too

not re-exporting from guild.ts right now, we can wait for others to see what they want to do

**Comment by vladfrangu** [comment]
> since users _can_ import individual files too

No, they cannot. So moving exports from the right file is perfectly fine

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**